pymc/tests/test_distributions_moments.py

```diff
@@ -1036,3 +1037,28 @@ def test_polyagamma_moment(h, z, size, expected):
     with Model() as model:
         PolyaGamma("x", h=h, z=z, size=size)
     assert_moment_is_expected(model, expected)
+
+
+@pytest.mark.parametrize(
+    "p, n, size, expected",
+    [
+        (np.array([0.25, 0.25, 0.25, 0.25]), 1, None, np.repeat(0, 4)),
```

Comment:
This is a bad moment, because it has a logp of `-np.inf`. We always need the moment to add up to `n`. IIRC there is no simple closed formula solution for the mode of the multinomial, but you should be able to use our hacky approximation from V3.

Reply:
This? https://github.com/pymc-devs/pymc/blob/c332f8c957bb2ebaf346832f440e5ee94b65026a/pymc3/distributions/multivariate.py#L583

Reply:
Yeah I think that's it. Basically it's a hack to make sure the values always add up to `n`, and not just in the case where they would all be zero

Reply:
Cool, I pretty much copied the code but made some changes for the cases when either `p` has more then 1 dimension or when `n` is an array rather than a scalar. Added tests for the corner cases I could think of. Once this PR is finalized, the same code can be copied to the DirichletMultinomial distribution!